services/visual-studio-marketplace/visual-studio-marketplace-version.service.js

```diff
@@ -27,15 +41,22 @@ export default class VisualStudioMarketplaceVersion extends VisualStudioMarketpl
     return renderVersionBadge({ version })
   }
 
-  transform({ json }) {
+  transform({ json }, includePrereleases) {
     const { extension } = this.transformExtension({ json })
-    const version = extension.versions[0].version
+    const preRelease = 'Microsoft.VisualStudio.Code.PreRelease'
+    const version = includePrereleases
+      ? extension.versions[0].version
+      : extension.versions.find(
+          obj => !obj.properties.find(({ key }) => key === preRelease)
+        ).version
+
```

Comment:
There are a couple of problems here.

1. If there are only pre-releases and no stable releases this will fail because `.find()` will just return `undefined`. For this case, we should fall back to showing the latest pre-release if there are no stable releases at all (this is what we normally do for other badges). 
2. This one is a bit more of an edge case: It looks like the deal is pre-releases are indicated by `{"key": "Microsoft.VisualStudio.Code.PreRelease", "value": "true"}` and stable releases are indicated by complete absence of that object, but I think if we were to encounter a release with `{"key": "Microsoft.VisualStudio.Code.PreRelease", "value": "false"}` we should also consider that stable (whether or not we currently think this is likely). This code would classify that as a pre-release.